### PR TITLE
[session2_4] 조영진 과제 제출합니다.

### PR DESCRIPTION
**Pull request by YOUNGJIN-CHO-1**
딥러닝은 저번 학기에도 지금도 여전히 어렵네요

그래도 어려운 내용을 잘 풀어서 설명해주셔서 조금은 이해할 수 있었습니다 

과제하면서 딥러닝 코드를 열심히 공부했는데 사용하려고 하면 또 기억이 안 날 거 같은 느낌입니다 ㅋㅋㅋ

딥러닝 공부는 엄두가 안 났었는데 이번을 계기로 한 번 제대로 공부해봐야겠네요

자료, 실습, 과제 어느 하나 빠질 것 없이 퀄리티가 좋네요

딥러닝 초심자가 딥러닝 공부하기에 아주 좋은 자료라고 생각합니다

감사합니다

### CODE REVIEW

**Comment by dookongee** [comment]
과제 정말 잘하셨어요~~
한가지 수정할 부분은
코드 채우는 부분에서 epoch = 3  총 3번의 에포크 동안 모델 학습하는 부분의 코드에
optimizer라고 하면 파라미터 업데이트가 안돼요. loss.backward()라고 하면 계산한 손실값을 이전 층에게 다 알려줘서, "이런 문제가 있어!"를 알려줘요. 그 다음에 optimizer.step()을 해야 loss.backward()로 알게 된 오류만큼 모델 파라미터 값을 수정해서 모델이 업데이트돼요. 
optimizer.step()을 안 하고, optimizer라고만 하면 손실은 계속 계산하지만 모델 성능이 나아지지 않아요. 그래서 Cost가 Epoch 1, 2, 3동안 계속 비슷한 2.3을 기록한 거에요. 
optimizer.step() 기억해주세요!! 

꼭 기억하실 부분은 DataSet과 DataLodaer를 사용해서 학습한다는 것과 model.train(), model.eval()이에요. 
Dataset을 통해 데이터들을 이미지-정답 pair로 구성해놓고, 이것을 DataLoader를 사용해서 한 번에 batch_size라고 지정한 개수만큼의 데이터를 하나의 묶음으로 가져와요. 

model을 설계하고 나서 학습하기 위해서는 model.train()을 통해 학습모드로 설정해주고, 
학습된 모델로 테스트를 할 때는 model.eval()를 통해 평가모드로 설정해야 해요

cnn이 아니고 다른 어떤 딥러닝을 하더라도 공통적으로 사용하는 내용이니까 꼭 기억해주세요!!
